fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(feature_config=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    feature_config : dict, default=None
+        A dict whose keys are the labels for the gender feature, and whose set
+        the :code:`n_samples` and :code:`class_sep` parameters for :code:`make_classification`.
+        If `None`, the following dictionary is used by default::
+
+            {
+                'Man': {'n_samples': 500, 'class_sep': 1},
+                'Other': {'n_samples': 500, 'class_sep': 0.5},
+                'Unspecified': {'n_samples': 500, 'class_sep': 0.5},
+                'Woman': {'n_samples': 500, 'class_sep': 2},
+            }
+
+    n_features : int, default=20
+        The total number of features.
+
+    n_informative : int, default=4
+        The number of informative features.
+
+    random_state : int or RandomState instance, default=None
+        The random number generator seed or :class:`numpy.random.RandomState` to use.
+
+    Returns
+    -------
+    (X, y, gender) : tuple of numpy.ndarray
+        X : numpy.ndarray
+            The generated samples.
+        y : numpy.ndarray
+            Labels for the binary classification.
+        gender : numpy.ndarray
+            The sensitive feature label.
+    """
+    rng = check_random_state(random_state)
+
+    classification_kwargs = {
+        'n_features': n_features,
+        'n_informative': n_informative,
+        'n_classes': 2,
+        'random_state': rng,
+    }
+
+    if feature_config is None:
+        feature_config = {
+            'Man': {'n_samples': 500, 'class_sep': 1},
+            'Other': {'n_samples': 500, 'class_sep': 0.5},
+            'Unspecified': {'n_samples': 500, 'class_sep': 0.5},
+            'Woman': {'n_samples': 500, 'class_sep': 2},
+        }
+
+    Xs, ys, genders = [], [], []
+
+    for label, group_config in feature_config.items():
+        group_config.update(classification_kwargs)
+        X, y = make_classification(**group_config)
+        genders.append([label] * group_config['n_samples'])
+        Xs.append(X)
+        ys.append(y)
+
+    X = np.concatenate(Xs)
+    y = np.concatenate(ys)
+    gender = np.concatenate(genders)
+
+    # Now generate disparities by randomly flipping classes
+    bias = {group_name: rng.uniform(0.4, 0.6) for group_name in feature_config}
```

Comment:
I see what you're doing here and it's a nice and simple way of achieving this, although it doesn't necessarily guarantee that we'll have any disparities, which might mean that users have to try a few times with different seeds. We could alternatively make this `bias` a part of the input args. I've added another comment where I suggest using a class for setting the config, so you can think of this as another arg to the config. I do think this could be another part of the follow-up PR, though.

Reply:
I've replaced this with a much better solution. See https://github.com/fairlearn/fairlearn/pull/907#discussion_r673120236

Reply:
do we need this? Different ways of calling `make_classification` is making that bias already.